src/WordSearch/WordSearchSolver.ts

```diff
@@ -0,0 +1,103 @@
+import trie = require('trie-prefix-tree');
+import {Point} from './Point';
+import {Result} from './Result';
+
+export class WordSearchSolver {
+  // The grid of letters which makes up the wordsearch puzzle
+  private _matrix: string[][];
+  // Transpositions to apply when finding words
+  private _directions: number[][];
+  // Potential words to find in the letter grid
+  private _targets: ReturnType<typeof trie>;
+
+  constructor(matrix: string[][]) {
+    this._matrix = matrix;
+
+    this._directions = [
+      // Cardinal directions
+      [0, 1], [-1, 0], [1, 0], [0, -1],
+      // Diagonal
+      [1, 1], [-1, -1], [1, -1], [-1, 1]
+    ];
+
+    this._targets = trie([]);
+  }
+
+  findWords(words: string[]): Result[] {
+    for (const word of words) {
+      if (word === null || typeof (word) === 'undefined') {

```

Comment:
You shouldn't need to do typeof, `word === undefined` should work fine

Reply:
Also a nit, `typeof` is an operator, not a function, so that's why the parens are weird, typically you wouldn't bother with them at all.